lib/mongo/retryable/base_worker.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+# encoding: utf-8
+
+# Copyright (C) 2015-2020 MongoDB Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+  module Retryable
+
+    # The abstract superclass for workers employed by Mongo::Retryable.
+    #
+    # @api private
+    #
+    # @since 2.19.0
+    class BaseWorker
```

Comment:
I assume that this class is not part of our public API. In this case we should mark it with `@api private`.